gtdynamics/dynamics/Chain.cpp

```diff
@@ -0,0 +1,102 @@
+/* ----------------------------------------------------------------------------
+ * GTDynamics Copyright 2021, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file Chain.cpp
+ * @brief Create a serial kinematic chain.
+ * @author Dan Barladeanu, Frank Dellaert.
+ */
+
+#include <gtdynamics/dynamics/Chain.h>
+
+namespace gtdynamics {
+
+Chain operator*(const Chain &chainA, const Chain &chainB) {
+  // expect 6 rows in jacobians
+  if (chainA.axes().rows() != 6 || chainB.axes().rows() != 6) {
+    throw std::runtime_error("Jacobians should have 6 rows in SE(3)");
+  }
+
+  // Compose poses:
+  Pose3 aTb = chainA.sMb();
+  Pose3 bTc = chainB.sMb();
+  Pose3 aTc = aTb.compose(bTc);
+
+  // Adjoint the first Jacobian to the new end-effector frame C:
+  Matrix c_Ad_b = bTc.inverse().AdjointMap();
+
+  // Multiply with first chain screw axes:
+  Matrix mult = c_Ad_b * chainA.axes();
+
+  // Create Matrix with correct number of columns:
+  Matrix out(mult.rows(), mult.cols() + chainB.axes().cols());
+
+  // Fill Matrix:
+  out << mult, chainB.axes();
+
+  return Chain(aTc, out);
+}
+
+Chain Chain::compose(std::vector<Chain> &chain_vector) {
+  // Initialize total chain
+  Matrix emptyMat(6, 0);
```

Comment:
Get rid of these 2, have a default constructor.

Reply:
done